packages/aws-cdk-lib/aws-cloudwatch/lib/variable.ts

```diff
@@ -0,0 +1,171 @@
+export enum VariableInputType {
+  /**
+   * Freeform text input box
+   */
+  INPUT = 'input',
+
+  /**
+   * A dropdown of pre-defined values, or values filled in from a metric search query
+   */
+  RADIO = 'radio',
+
+  /**
+   * A set of pre-defined radio buttons, which can also be defined from a metric search query
+   */
+  SELECT = 'select',
+}
+
+export enum VariableType {
+  /**
+   * A property variable changes the values of all instances of a property in the list of widgets in the dashboard.
+   */
+  PROPERTY = 'property',
+
+  /**
+   * A pattern variable is one that changes a regex pattern across the dashboard JSON
+   */
+  PATTERN = 'pattern',
+}
+
+/**
+ * A single dashboard variable
+ */
+export interface IVariable {
+  /**
+   * Return the variable JSON for use in the dashboard
+   */
+  toJson(): any;
+}
+
+export interface IDashboardVariable {
+  /**
+   * Type of the variable
+   */
+  readonly type: VariableType;
+
+  /**
+   * The way the variable value is selected
+   */
+  readonly inputType: VariableInputType;
+
+  /**
+   * Pattern or property value to replace
+   */
+  readonly value: string;
+
+  /**
+   * Unique id
+   */
+  readonly id: string;
+
+  /**
+   * Optional label in the toolbar
+   */
+  readonly label?: string;
+
+  /**
+   * Optional default value
+   */
+  readonly defaultValue: any;
+
+  /**
+   * Whether the variable is visible
+   */
+  readonly visible?: boolean;
+}
+
+export abstract class DashboardVariable implements IVariable {
+  private readonly baseProps: IDashboardVariable;
+
+  protected constructor(props: IDashboardVariable) {
+    this.baseProps = props;
+  }
+
+  toJson(): any {
+    return {
+      [this.baseProps.type]: this.baseProps.value,
+      type: this.baseProps.type,
+      inputType: this.baseProps.inputType,
+      id: this.baseProps.id,
+      defaultValue: this.baseProps.defaultValue,
+      visible: this.baseProps.visible,
+      label: this.baseProps.label,
+    };
+  }
+
+}
+
+export interface IVariableValue {
+  /**
+   * Optional label for the selected item
+   *
+   * @default - value
```

Comment:
What does `value` mean here? It's kind of an oversaturated word... if you mean the value you specify in the 'value' prop, then lets be specific :)